tools/lisa-build-asset

```diff
@@ -60,22 +70,54 @@ def list_assets(recipe_dir):
 def get_asset_recipe(asset, recipe_dir):
     return os.path.join(recipe_dir, '{}.recipe'.format(asset))
 
-def make(asset, actions, arch, toolchain, recipe_dir, build_dir):
+def get_recipe_name(asset):
+    return asset + ".recipe"
+
+def make_native_build(asset, arch, recipe_dir, build_dir, alpine_source_dir, arch_chroot_dir, env):
+    # Install the chroot
+    subprocess.check_call(['sudo', os.path.join(alpine_source_dir, 'alpine-chroot-install'),
+            '-d', arch_chroot_dir, '-i', build_dir, '-a', QEMU_ARCH_NAMES[arch],
+            '-p', ' '.join(ALPINE_PKG),
+            '-k', ' '.join(env['LISA_ASSET_RECIPE_USED_ENV'].split())],
+            cwd=build_dir)
+
+    # Copy the recipe where the sources are and append the function to execute
+    shutil.copyfile(get_asset_recipe(asset, recipe_dir), os.path.join(build_dir, asset + ".recipe"))
+    with open(os.path.join(build_dir, get_recipe_name(asset)), "a") as f:
+        f.write("build\n")
+
+    # Enter the chroot and build
+    subprocess.check_call([os.path.join(arch_chroot_dir, 'enter-chroot'),
+            '-u', os.environ['USER'], 'source', get_recipe_name(asset)],
+            env=env, cwd=build_dir)
+
+def make(asset, actions, arch, toolchain, recipe_dir, build_dir, arch_chroot_dir=None, alpine_source_dir=None):
+    native_build = alpine_source_dir and asset in NATIVE_BUILD and arch in QEMU_ARCH_NAMES
+    if native_build:
+        # Since the build is native, no need for a toolchain
+        toolchain = None
+
     recipe = get_asset_recipe(asset, recipe_dir)
     env = get_env(asset, arch, toolchain, recipe, build_dir)
     for action in actions:
         print('\nMaking {} {} {} in: {}\n'.format(arch, asset, action, build_dir))
 
+        if native_build and action == 'build':
+            make_native_build(asset, arch, recipe_dir, build_dir, alpine_source_dir, arch_chroot_dir, env)
+            continue
+
         script = 'source {recipe} && {action}'.format(
             recipe=recipe,
             action=action,
         )
         cmd = ['bash', '-c', script]
         subprocess.check_call(cmd, env=env, cwd=build_dir)
 
-ARCHITECTURES = ['arm64', 'armeabi', 'x86_64']
+        if native_build and action == 'install':
+            # Clean-up the chroot
+            subprocess.check_call([os.path.join(arch_chroot_dir, 'destroy'), '--remove'])
```

Comment:
should be handled with a context manager like the temporary build dir in `make_asset()`.
Using `contextlib.contextmanager`, you can safely encapsulate the chroot creation and teardown, so that it will always be cleaned up properly no matter what exception is raised in the middle: https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager

It also seems that `make_asset()` is the right level to use that context manager, and `make()` will therefore only need to take the `chroot` parameter when appropriate to run the action inside the chroot:

```
def make_asset(asset, arch_list, build_dir, recipe_dir, toolchains, native):
    ... 
    for arch in arch_list:
        if native:
            arch_chroot_dir  = os.path.join(arch_build_dir, 'chroot')
            chroot_cm = make_chroot(arch_chroot_dir)
        else: 
             arch_chroot_dir = None
             chroot_cm = nullcontext(None)
         ...
         make(...., chroot=arch_chroot_dir)
```